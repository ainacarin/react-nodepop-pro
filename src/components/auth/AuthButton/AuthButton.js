import { Link } from 'react-router-dom';
import T from 'prop-types';

import { ConfirmationButton } from '../../shared';
import { logout } from '../../../api/auth';
import { connect } from 'react-redux';
import { authLogout, authLogoutAction } from '../../../store/actions/auth';
import { getIsLogged } from '../../../store/selectors/auth';

const AuthButton = ({ isLogged, onLogout }) => {


  return isLogged ? (
    <ConfirmationButton
      confirmation="Are you sure?"
      onConfirm={onLogout}
    >
      Logout
    </ConfirmationButton>
  ) : (
    <Link to="/login">Login</Link>
  );
};

AuthButton.propTypes = {
  onLogout: T.func.isRequired,
  isLogged: T.bool,
};

AuthButton.defaultProps = {
  isLogged: false,
};

const mapDispatchToProps = (dispatch, ownProps) => ({
  onLogout: () => dispatch(authLogoutAction()),
});
const mapStateToProps = (state, ownProps) => ({ isLogged: getIsLogged(state) });
export default connect(mapStateToProps, mapDispatchToProps)(AuthButton);

export const AuthButtonTest = AuthButton;