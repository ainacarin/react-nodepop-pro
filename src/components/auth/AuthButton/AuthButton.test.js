import React from 'react';
import { shallow } from 'enzyme';

import { AuthButtonTest } from './AuthButton';

describe('AuthButtonTest', () => {
  const props = {
    isLogged: true,
    onLogout: jest.fn()
  };

  const render = () => shallow(<AuthButtonTest {...props} />);
  test('should render', () => {
    const wrapper = render();
    expect(wrapper.exists()).toBe(true);
  });

  test('snapshot testing', () => {
    const wrapper = render();
    expect(wrapper).toMatchSnapshot();
  });

  test('snapshot find element ConfirmationButton', () => {
    const wrapper = render();
    const confirmationButton = wrapper.find('ConfirmationButton');
    confirmationButton.props().onConfirm();
    expect(props.onLogout).toHaveBeenCalled()
  });

});
