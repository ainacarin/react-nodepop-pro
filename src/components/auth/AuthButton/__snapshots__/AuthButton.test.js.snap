// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthButtonTest snapshot testing 1`] = `
<ConfirmationButton
  confirmation="Are you sure?"
  onConfirm={[MockFunction]}
>
  Logout
</ConfirmationButton>
`;
